modules/mauth-authenticator/src/main/java/com/mdsol/mauth/RequestAuthenticator.java

```diff
@@ -152,7 +152,7 @@ private boolean validateSignatureV2 (MAuthRequest mAuthRequest, PublicKey client
         MAuthSignatureHelper.generateStringToSign(
             mAuthRequest.getAppUUID(), mAuthRequest.getHttpMethod(), mAuthRequest.getResourcePath(),
             mAuthRequest.getQueryParameters(),
-            new String(mAuthRequest.getMessagePayload(), StandardCharsets.UTF_8),
+            mAuthRequest.getMessagePayload(),
             String.valueOf(mAuthRequest.getRequestTime()),
```

Comment:
If this is always a `long` the function should take a `long` instead of `String` and move the long -> String conversion into `generateStringToSign`. The general principle is to make it hard for users of your API to screw things up

Reply:
It is just follow the original interface of **generateUnencryptedSignature**, easy for the current user do upgrade?

Reply:
sure I guess this interface is also easy when you want to validate an incoming request (since all headers are strings)